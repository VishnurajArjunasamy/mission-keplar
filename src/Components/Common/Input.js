import React from "react";

export default function Input({
  className,
  type,
  onKeyDown,
  value,
  placeholder,
}) {
  return (
    <input
      className={className}
      type={type}
      value={value}
      onKeyDown={onKeyDown}
      placeholder={placeholder}
    />
  );
}
