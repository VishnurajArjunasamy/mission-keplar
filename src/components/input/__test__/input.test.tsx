import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import Input from "../input";
import { Provider } from "react-redux";

const mockStore = configureStore([]);

describe("Input Component", () => {
  let store: ReturnType<typeof mockStore>;

  beforeEach(() => {
    store = mockStore({
      sideBar: {
        isDarkMode: false,
      },
    });
  });

  it("renders input with light mode styles", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Input
          type="text"
          value="Test Value"
          placeholder="Test Placeholder"
          name="testName"
          isReadOnly={false}
        />
      </Provider>
    );
    const input = screen.getByPlaceholderText("Test Placeholder");
    expect(input).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
    expect(input).toHaveClass("light");
    expect(input).not.toHaveClass("dark");
  });

  it("renders input with dark mode styles", () => {
    store = mockStore({
      sideBar: {
        isDarkMode: true,
      },
    });
    const { asFragment } = render(
      <Provider store={store}>
        <Input
          type="text"
          value="Test Value"
          placeholder="Test Placeholder"
          name="testName"
          isReadOnly={false}
        />
      </Provider>
    );
    const input = screen.getByPlaceholderText("Test Placeholder");
    expect(input).toBeInTheDocument();
    expect(input).not.toHaveClass("light");
    expect(input).toHaveClass("dark");
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with provided props", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Input
          type="text"
          value="Test Value"
          placeholder="Test Placeholder"
          name="testName"
          isReadOnly={false}
        />
      </Provider>
    );

    const input = screen.getByPlaceholderText("Test Placeholder");
    expect(input).toBeInTheDocument();
    expect(input).toHaveAttribute("type", "text");
    expect(input).toHaveAttribute("value", "Test Value");
    expect(input).toHaveAttribute("name", "testName");
    expect(input).not.toHaveAttribute("readonly");
    expect(asFragment()).toMatchSnapshot();
  });

  it("handles onChange events", () => {
    const { asFragment } = render(
      <Provider store={store}>

      <Input
        type="text"
        value=""
        placeholder="Test Placeholder"
        onChange={jest.fn()}
      /></Provider>
    );

    const input = screen.getByPlaceholderText("Test Placeholder");
    fireEvent.change(input, { target: { value: "New Value" } });

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies default value correctly", () => {
    const { asFragment } = render(
      <Provider store={store}>

      <Input
        type="text"
        defaultValue="Default Value"
        placeholder="Test Placeholder"
      /></Provider>
    );

    const input = screen.getByPlaceholderText("Test Placeholder");
    expect(input).toHaveAttribute("value", "Default Value");
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders as read-only when isReadOnly is true", () => {
    const { asFragment } = render(
      <Provider store={store}>
      <Input
        type="text"
        isReadOnly={true}
        placeholder="Read-Only Placeholder"
      /></Provider>
    );

    const input = screen.getByPlaceholderText("Read-Only Placeholder");
    expect(input).toHaveAttribute("readonly");
    expect(asFragment()).toMatchSnapshot();
  });

  it("forwards ref to the input element", () => {
    const inputRef = React.createRef<HTMLInputElement>();
    const { asFragment } = render(
      <Provider store={store}>
        <Input type="text" ref={inputRef} placeholder="Test Ref Placeholder" />
      </Provider>
    );

    expect(inputRef.current).toBeInstanceOf(HTMLInputElement);
    expect(inputRef.current?.placeholder).toBe("Test Ref Placeholder");
    expect(asFragment()).toMatchSnapshot();
  });
});
