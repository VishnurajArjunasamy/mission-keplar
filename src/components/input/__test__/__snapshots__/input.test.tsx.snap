// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component applies default value correctly 1`] = `
<DocumentFragment>
  <input
    class="input light"
    placeholder="Test Placeholder"
    type="text"
    value="Default Value"
  />
</DocumentFragment>
`;

exports[`Input Component forwards ref to the input element 1`] = `
<DocumentFragment>
  <input
    class="input light"
    placeholder="Test Ref Placeholder"
    type="text"
  />
</DocumentFragment>
`;

exports[`Input Component handles onChange events 1`] = `
<DocumentFragment>
  <input
    class="input light"
    placeholder="Test Placeholder"
    type="text"
    value=""
  />
</DocumentFragment>
`;

exports[`Input Component renders as read-only when isReadOnly is true 1`] = `
<DocumentFragment>
  <input
    class="input light"
    placeholder="Read-Only Placeholder"
    readonly=""
    type="text"
  />
</DocumentFragment>
`;

exports[`Input Component renders correctly with provided props 1`] = `
<DocumentFragment>
  <input
    class="input light"
    name="testName"
    placeholder="Test Placeholder"
    type="text"
    value="Test Value"
  />
</DocumentFragment>
`;

exports[`Input Component renders input with dark mode styles 1`] = `
<DocumentFragment>
  <input
    class="input dark"
    name="testName"
    placeholder="Test Placeholder"
    type="text"
    value="Test Value"
  />
</DocumentFragment>
`;

exports[`Input Component renders input with light mode styles 1`] = `
<DocumentFragment>
  <input
    class="input light"
    name="testName"
    placeholder="Test Placeholder"
    type="text"
    value="Test Value"
  />
</DocumentFragment>
`;
